crashlogs.cpp

```diff
@@ -153,23 +172,59 @@ namespace glaiel::crashlogs {
         cv.wait(lk, [] { return status != program_status::crashed; });
     }
 
+    //Try to get the string representation of a signal identifier, return an empty string if none is found.
+    //This only covers the signals from the C++ std lib and none of the POSIX or OS specific signal names!
+    static std::string_view try_get_signal_name(int signal) {
+        switch (signal)
+        {
+            case SIGTERM:
+                return "SIGTERM";
+            case SIGSEGV:
+                return "SIGSEGV";
+            case SIGINT:
+                return "SIGINT";
+            case SIGILL:
+                return "SIGILL";
+            case SIGABRT:
+                return "SIGABRT";
+            case SIGFPE:
+                return "SIGFPE";
+        }
+        return "";
+    }
+
     //various callbacks needed to get into the crash handler during a crash (borrowed from backward.cpp)
     static inline void signal_handler(int signal) {
+        std::stringstream text;
+        text << "Received signal " << signal << " " << try_get_signal_name(signal);
+
+        std::cout << text.rdbuf() << std::endl;
+        header_reason = text.str();
+
         crash_handler();
-        abort();
+        std::quick_exit(1);
     }
+
     static inline void terminator() {
+        std::cout << "Application terminated" << std::endl;
+        header_reason = "Application terminated";
+
         crash_handler();
-        abort();
+        std::quick_exit(1);
     }
+#ifdef OS_WINDOWS
     __declspec(noinline) static LONG WINAPI exception_handler(EXCEPTION_POINTERS* info) {
+        //TODO consider writing additional output from info to header_reason
         crash_handler();
         return EXCEPTION_CONTINUE_SEARCH;
     }
     static void __cdecl invalid_parameter_handler(const wchar_t*,const wchar_t*,const wchar_t*,unsigned int,uintptr_t) {
+        //TODO consider writing additional output from info to header_reason

```

Comment:
would need to not do work, again should *just* write an integer global value and nothing else